azure/azure_kv.go

```diff
@@ -62,26 +69,34 @@ func New(
 		return nil, ErrAzureClientIDNotSet
 	}
 	secretID := getAzureKVParams(secretConfig, AzureClientSecret)
-	if secretID == "" {
-		return nil, ErrAzureSecretIDNotSet
-	}
-	envName := getAzureKVParams(secretConfig, AzureEnviornment)
-	if envName == "" {
-		// we set back to default AzurePublicCloud
-		envName = AzureCloud
-	}
+	clientCertPath := getAzureKVParams(secretConfig, AzureClientCertPath)
+	clientCertPassword := getAzureKVParams(secretConfig, AzureClientCertPassword)
+
 	vaultURL := getAzureKVParams(secretConfig, AzureVaultURL)
 	if vaultURL == "" {
 		return nil, ErrAzureVaultURLNotSet
 	}
 
-	client, err := getAzureVaultClient(clientID, secretID, tenantID, envName)
-	if err != nil {
-		return nil, err
+	clientOpts := getAzureClientOptions(secretConfig)
+
+	var client *azsecrets.Client
+	var err error
+	if secretID != "" {
+		client, err = getAzureVaultClient(clientID, secretID, tenantID, vaultURL, clientOpts)
+		if err != nil {
+			return nil, err
+		}
+	} else if clientCertPath != "" {
+		client, err = getAzureVaultClientWithCert(clientID, tenantID, vaultURL, clientCertPath, clientCertPassword, clientOpts)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
It seems to me like the libopenstorage/secrets library should be the place where this logic happens. That would allow us to directly pass user config values through to the library without needing to understand how the library works.

Reply:
marking it as resolved after the offline discussion.